packages/wrangler/pages/functions/filepath-routing.ts

```diff
@@ -28,101 +26,60 @@ export async function generateConfigFromFileTree({
   await forEachFile(baseDir, async (filepath) => {
     const ext = path.extname(filepath);
     if (/^\.(mjs|js|ts|tsx|jsx)$/.test(ext)) {
-      // transform the code to ensure we're working with vanilla JS + ESM
-      const { code } = await transform(await fs.readFile(filepath, "utf-8"), {
-        loader: ext === ".ts" ? "ts" : "js",
+      const { metafile } = await build({
+        metafile: true,
+        write: false,
+        entryPoints: [path.resolve(filepath)],
       });
+      const exportNames: string[] = [];
+      if (metafile) {
+        for (const output in metafile?.outputs) {
```

Comment:
What can `metafile.outputs` contain? If it is only the bundled file then there should only be one, right? If so, then I think we are good. If it can also contain all the input files that have been transpiled, then you might end up with false positives here.

Reply:
Looks like it's only the file in the `entrypoints` that's processed here.
I'll add `bundle: false` to the build command so it's clearer that we only care about that single file's exports.

Here's an example file:
```typescript
import { onRequestDelete } from "./bar";
export function onRequestGet() {}
export function onRequestPost() {}
export async function foo() {
    return onRequestDelete()
}
```

Metafile:
```json
{
  "inputs": {
    "../../tmp/functions/foo.ts": {
      "bytes": 171,
      "imports": []
    }
  },
  "outputs": {
    "foo.js": {
      "imports": [],
      "exports": [
        "foo",
        "onRequestGet",
        "onRequestPost"
      ],
      "entryPoint": "../../tmp/functions/foo.ts",
      "inputs": {
        "../../tmp/functions/foo.ts": {
          "bytesInOutput": 172
        }
      },
      "bytes": 172
    }
  }
}
```

Reply:
Great.